cmd/oras/internal/display/print.go

```diff
@@ -87,8 +87,28 @@ func NewTagManifestStatusPrinter(target oras.Target) oras.Target {
 	}
 }
 
```

Comment:
Yes, agreed on separating of concerns. Also that's why I use `hintFunc` to let callee define the output content. 

Logrus currently is used for structured logging to stderr, for stdout output, oras don't have specific design or utility for logging. I think for me there are still some gaps on getting the overall vision of #786, it would be great if you can provide some opensource projects that have similar implementation.

Reply:
I wasn't thinking about something to replace logrus, just some code shuffled around there. I also feel this current architecture would impeded the usefulness fromhttps://github.com/oras-project/oras/pull/825 the go script PR

Reply:
>I also feel this current architecture would impeded the usefulness fromhttps://github.com/https://github.com/oras-project/oras/pull/825 the go script PR

Yes, you are right. The current of output is designed to read by human, providing as much information as possible. The drawback is clear: the logs are not structured and very hard to parsed.

Reply:
Also, to facilitate go scripting, oras should provide a new output mode like mentioned in https://github.com/oras-project/oras/issues/638 to output a single structured data for every execution.